core/src/main/scala/spinal/core/Enum.scala

```diff
@@ -82,14 +82,21 @@ class SpinalEnum(var defaultEncoding: SpinalEnumEncoding = native) extends Namea
 
   /** Contains all elements of the enumeration */
   @dontName val elements = ArrayBuffer[SpinalEnumElement[this.type]]()
+  @dontName var encodingMap = Map[BigInt, SpinalEnumElement[this.type]]()
```

Comment:
This is problematic, basicaly the SpinalEnum class is a kind of template which has a recerence to a default encoding.

Only encodings themself know the mapping between a given element and its hardware values, not the SpinalEnum itself.

I will give a look now